### PR TITLE
Catcher streamline

### PR DESCRIPTION
**Pull request by AaronParsons**
Couple more changes to control restructuring, and a vital change to CATCHER BUFFER size that gets rid of packet loss.

### CODE REVIEW

**Review by plaplant — CHANGES_REQUESTED** [review]
Thanks @AaronParsons! Glad to hear that this solves the traffic bottleneck on site. What was the specific change that fixed things? It looks like some of the packet/block math changed, but I'm curious what was most important.

Most of my comments are about removing commented out code rather can keeping it around. Though I appreciate all of the explanatory comments, they are very helpful.

File: control_software/scripts/hera_catcher_up.py [hunk 1]
```diff
@@ -33,7 +33,11 @@
 # Environment sourcing command required to run remote python jobs
 python_source_cmd = ["source", os.path.join(args.pypath, "bin/activate"), "hera", ";"]
 
-r = redis.Redis(args.redishost, decode_responses=True)
+#r = redis.Redis(args.redishost, decode_responses=True)
+#r = redis.Redis(args.redishost)
+#sub = r.pubsub()
+#sub.subscribe('hashpipe://%s/%d/set' % (args.host, 0))
+
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply control_software/scripts/hera_catcher_up.py:40]: This was a partial roll out of better handshaking to get away from magical sleeps, but it doesn't work yet. Let's keep it for now to see if I can get this working later.

File: control_software/scripts/hera_catcher_up.py [hunk 2]
```diff
@@ -45,16 +49,24 @@
...
 
 catcher.clear_redis_keys(args.redishost)
 
+# Wait for threads to be ready
+#for msg in sub.listen():
+#    print(msg)
+#    if msg['type'] == 'message' and msg['data'] == 'CNETHOLD=0':
+#        break
+#    else:
+#        continue
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply control_software/scripts/hera_catcher_up.py:68]: Same answer as above.

File: control_software/scripts/hera_catcher_up.py [hunk 3]
```diff
@@ -45,16 +49,24 @@
 
 # Start Catcher
 run_on_hosts([args.host], python_source_cmd + ['cd', '/data;', 'hera_catcher_init.sh'] + init_args + ['0'], wait=True)
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply control_software/scripts/hera_catcher_up.py:51]: Er... aren't we?

File: src/hera_catcher_net_thread.c [hunk 4]
```diff
@@ -110,13 +111,19 @@ static uint32_t set_block_filled(hera_catcher_bda_input_databuf_t *db, block_inf
   uint64_t block_missed_pkt_cnt;
   uint64_t block_missed_xengs, block_missed_mod_cnt, missed_pkt_cnt=0;
   uint32_t block_i = block_for_bcnt(binfo->bcnt_start);
-  int i;
+  //int i;
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply src/hera_catcher_net_thread.c:114]: It's attached to a block of error prints that I commented out but would like to fix so they actually work. When that happens, we can uncomment here.

File: src/hera_catcher_net_thread.c [hunk 5]
```diff
@@ -110,13 +111,19 @@ static uint32_t set_block_filled(hera_catcher_bda_input_databuf_t *db, block_inf
...
+  // ARP don't understand last_filled math here (isn't it -1?). Also, this should work so
+  // it is impossible to screw up the order.
+  // looks like it is implicitly tied to starting at block_i=0, and then advances
+  // along with it, as long as set_block_filled is only called once per block
+  // but if we skip a block, we never set filled, and the disk writer will lock
   last_filled = (last_filled+1) % CATCHER_N_BLOCKS;
   if(last_filled != block_i) {
     printf("block %d being marked filled, but expected block %d!\n", block_i, last_filled);
   }
+  printf("net_thread: marking block_i=%d filled\n", block_i);
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply src/hera_catcher_net_thread.c:126]: It does, and I'll take it out later, but for now, I need visibility into the microscopic stuff if things break...

File: src/hera_catcher_net_thread.c [hunk 6]
```diff
@@ -153,11 +160,12 @@ static uint32_t set_block_filled(hera_catcher_bda_input_databuf_t *db, block_inf
...
-    for (i=0; i<N_XENGINES; i++){
-      fprintf(stderr, "XengID %2d: %.2f\n", i,
-              (float)binfo->xeng_pkt_counter[block_i][i]/PACKETS_PER_X);
-    }
+    //fprintf(stderr, "Fraction pkts received:\n");
+    //for (i=0; i<N_XENGINES; i++){
+    //  // ARP: this print is numerically incorrect. xeng_pkt_counter is never incremented
+    //  fprintf(stderr, "XengID %2d: %.2f\n", i,
+    //          (float)binfo->xeng_pkt_counter[block_i][i]/PACKETS_PER_X);
+    //}
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply src/hera_catcher_net_thread.c:168]: I agree (as per above), but I'm not ready to sink any more time in this right now. Can we kick this down the road?

File: control_software/scripts/hera_catcher_up.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AaronParsons) [reply control_software/scripts/hera_catcher_up.py:51]: It looks to me like run_on_hosts is imported from that module.

**Comment by AaronParsons** [comment]
I hear what you're saying on these comments wrt cleanup, but it mostly reflects work half-done that I'd like to lock in before proceeding further. Since last reveiw, I did add 1 more commit to fix the BDANANT bug that i think is freezing the autocorr thread.

**Review by plaplant — APPROVED** [review]

**Pull request merged**